app/controllers/sprangular/locale_controller.rb

```diff
@@ -0,0 +1,14 @@
+module Sprangular
+  ##
+  # API Endpoint to set locale for session
+  class LocaleController < Sprangular::BaseController
+    def show
+      session[:locale] = params[:locale]
+
+      respond_to do |format|
+        format.js { render js: 'window.location.reload' }
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.